bindings/wasm/src/credential/revocation/status_list_2021/credential.rs

```diff
@@ -0,0 +1,251 @@
+      CredentialStatus::Suspended => Self::Suspended,
+      CredentialStatus::Valid => Self::Valid,
+    }
+  }
+}
+
+impl From<WasmCredentialStatus> for CredentialStatus {
+  fn from(value: WasmCredentialStatus) -> Self {
+    match value {
+      WasmCredentialStatus::Revoked => Self::Revoked,
+      WasmCredentialStatus::Suspended => Self::Suspended,
+      WasmCredentialStatus::Valid => Self::Valid,
+    }
+  }
+}
+
+/// Purpose of a {@link StatusList2021}.
+#[wasm_bindgen(js_name = StatusPurpose)]
+#[derive(Clone, Copy, Debug, Eq, PartialEq, Hash)]
+pub enum WasmStatusPurpose {
+  Revocation = 0,
+  Suspension = 1,
+}
+
+impl From<StatusPurpose> for WasmStatusPurpose {
+  fn from(value: StatusPurpose) -> Self {
+    match value {
+      StatusPurpose::Revocation => Self::Revocation,
+      StatusPurpose::Suspension => Self::Suspension,
+    }
+  }
+}
+
+impl From<WasmStatusPurpose> for StatusPurpose {
+  fn from(value: WasmStatusPurpose) -> Self {
+    match value {
+      WasmStatusPurpose::Revocation => Self::Revocation,
+      WasmStatusPurpose::Suspension => Self::Suspension,
+    }
+  }
+}
+
+/// A parsed [StatusList2021Credential](https://www.w3.org/TR/2023/WD-vc-status-list-20230427/#statuslist2021credential).
+#[wasm_bindgen(js_name = "StatusList2021Credential", inspectable)]
+#[derive(Clone, Debug, Eq, PartialEq, Serialize, Deserialize)]
+#[serde(from = "StatusList2021Credential", into = "StatusList2021Credential")]
+pub struct WasmStatusList2021Credential {
+  pub(crate) inner: StatusList2021Credential,
+  wasm_credential: WasmCredential,
+}
+
+impl Deref for WasmStatusList2021Credential {
+  type Target = WasmCredential;
+  fn deref(&self) -> &Self::Target {
+    &self.wasm_credential
+  }
+}
+
+impl From<StatusList2021Credential> for WasmStatusList2021Credential {
+  fn from(value: StatusList2021Credential) -> Self {
+    Self {
+      wasm_credential: WasmCredential(value.clone().into_inner()),
+      inner: value,
+    }
+  }
+}
+
+impl From<WasmStatusList2021Credential> for StatusList2021Credential {
+  fn from(value: WasmStatusList2021Credential) -> Self {
+    value.inner
+  }
+}
+
+#[wasm_bindgen(js_class = StatusList2021Credential)]
+impl WasmStatusList2021Credential {
+  /// Creates a new {@link StatusList2021Credential}.
+  #[wasm_bindgen(constructor)]
+  pub fn new(credential: WasmCredential) -> Result<WasmStatusList2021Credential> {
+    StatusList2021Credential::try_from(credential.0)
+      .map(Into::into)
+      .map_err(|e| JsValue::from(JsError::new(&e.to_string())))
+  }
+
+  #[wasm_bindgen]
+  pub fn id(&self) -> String {
+    self.inner.id.as_deref().map(ToString::to_string).unwrap()
+  }
+
+  /// Sets the given credential's status using the `index`-th entry of this status list.
+  /// Returns the created `credentialStatus`.
+  #[wasm_bindgen(js_name = "setCredentialStatus")]
+  pub fn set_credential_status(
+    &mut self,
+    credential: &mut WasmCredential,
+    index: usize,
+    flag: bool,
```

Comment:
in Rust we call in value, is there a reason to differentiate?

Reply:
Changed this to `value`.